app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class MoviesController < ApplicationController
+    def index 
+        @movies = Movie.all 
+        @movies = @movies.where('title like ?', "%#{params[:title]}%") if params[:title]
+
+        render json: @movies
```

Comment:
https://stackoverflow.com/a/10280944

If you don't share variables between different methods, they should be local variables, so without `@`.